### PR TITLE
Added spree globalize support

### PR DESCRIPTION
**Pull request by drkpkg**
Based in the solution from the issue response https://github.com/spree-contrib/spree_multi_vendor/issues/104#issuecomment-614384117 I added the migrations and decorators only if SpreeGlobalize is defined.

### CODE REVIEW

File: app/models/spree/vendor_translation_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  if defined? (SpreeGlobalize)
+    def self.prepended(base)
+      base.translates :name, :about_us, :contact_us, :slug, fallbacks_for_empty_translations: true
+    end
+
+    Spree::Vendor.include SpreeGlobalize::Translatable
+  end
+end
+
+::Spree::Vendor.prepend(Spree::VendorDecorator)
```
Comment (damianlegawiec) [inline app/models/spree/vendor_translation_decorator.rb]:
```suggestion
::Spree::Vendor.prepend(Spree::VendorDecorator) if defined? (SpreeGlobalize)
```

File: app/models/spree/vendor_translation_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Spree::VendorDecorator
+  if defined? (SpreeGlobalize)
```
Comment (damianlegawiec) [inline app/models/spree/vendor_translation_decorator.rb]:
```suggestion

```

File: app/models/spree/vendor_translation_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    def self.prepended(base)
+      base.translates :name, :about_us, :contact_us, :slug, fallbacks_for_empty_translations: true
```
Comment (damianlegawiec) [inline app/models/spree/vendor_translation_decorator.rb]:
```suggestion
      if defined? (SpreeGlobalize)
        base.translates :name, :about_us, :contact_us, :slug, fallbacks_for_empty_translations: true
      end
```

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

**Review by drkpkg — COMMENTED** [review]
More changes :)

File: app/models/spree/vendor_translation_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+module Spree::VendorDecorator
+    def self.prepended(base)
+      if defined? (SpreeGlobalize)
+        base.translates :name, :about_us, :contact_us, :slug, fallbacks_for_empty_translations: true
+      end
+    end
+
+    Spree::Vendor.include SpreeGlobalize::Translatable
```
Comment (damianlegawiec) [inline app/models/spree/vendor_translation_decorator.rb]: @drkpkg you can actually put this code in https://github.com/spree-contrib/spree_multi_vendor/blob/master/app/models/spree/vendor.rb

Comment (damianlegawiec) [inline app/models/spree/vendor_translation_decorator.rb]:
```suggestion
    Spree::Vendor.include SpreeGlobalize::Translatable if defined?(SpreeGlobalize)
```

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**